core/src/test/scala/unit/kafka/utils/TestUtils.scala

```diff
@@ -441,6 +441,17 @@ object TestUtils extends Logging {
       replicaAssignment.size
     }
 
+    def isTopicExistsAndHasSameNumPartitionsAndReplicationFactor(cause: Throwable): Boolean = {
+      // return true when:
+      //  (1) the cause of exception is `TopicExistsException`
+      //  (2) all partitions metadata are propagated
+      //  (3) topic has the expected partitions number and replication factor
```

Comment:
Is this comment adding anything over the code?

Reply:
I agree that the comments are restating the code, but one thing that may be useful to note (not clear from the method name) is that we are waiting for the partition metadata now in this method.

Reply:
Agree. Adding one comment:
```
wait until all partitions metadata are propagated before verifying partitions number and replication factor
```
and remove the others.
I'll merge it tomorrow. Thanks.